src/catalog/src/durable/persist.rs

```diff
@@ -157,17 +158,48 @@ impl FenceableEpoch {
             } => Some(current_epoch.clone()),
         }
     }
+
+    /// Returns `Err` if `epoch` fences out `self`, `Ok` otherwise.
+    fn maybe_fence(&mut self, epoch: Epoch) -> Result<(), DurableCatalogError> {
+        match self {
+            FenceableEpoch::Unfenced(Some(current_epoch)) => {
+                if epoch > *current_epoch {
+                    *self = FenceableEpoch::Fenced {
+                        current_epoch: *current_epoch,
+                        fence_epoch: epoch,
+                    };
+                    self.validate()?;
+                } else if epoch < *current_epoch {
+                    panic!("Epoch went backwards from {current_epoch:?} to {epoch:?}");
```

Comment:
this scares me a bit, but it already exists so I guess it's not too big of a deal?

Reply:
Yeah it already existed. We could change this from `panic!` to `error!`, I just worry that we'll be in some undefined state and may cause damage to the catalog if we try to continue.